src/scripts/main.js

```diff
@@ -1,3 +1,23 @@
 'use strict';
 
-// write code here
+const employees = Array.from(document.querySelectorAll('ul li'));
+
+employees.sort((a, b) => {
+  const salaryA = parseFloat(
+    a.getAttribute('data-salary').replace(/[^\d.-]/g, ''),
```

Comment:
Ensure that the 'data-salary' attribute is always present and correctly formatted as a number. If there's a possibility of missing or malformed data, consider adding error handling or default values to prevent NaN results.